### DISCUSSION TITLE
[Support] interface.ts 中对@@midwayjs/core的Context的定义突然不生效了

### DISCUSSION THREAD

**Discussion by minefujiko9293** [midwayjs/midway#3508] [Q&A] [answered]
### Describe the problem（描述问题）

文档 [https://www.midwayjs.org/docs/context_definition](url) 扩展上下文定义。

之前项目是在**V3.12.X**版本下构建的，在interface.ts中拓展了Context的定义：
```
// interface.ts

import '@midwayjs/core';
import type {} from '@midwayjs/logger';
import { NW_ADMIN_SYS_USER } from './entity/source1/local/NW_ADMIN_SYS_USER.entity';

declare module '@midwayjs/core' {
  interface MidwayConfig {
    myConfig?: {
      minioBucketName?: string;
      hosts?: {
        bj_auth: string;
        bj_point: string;
      };
    };
  }

  interface Context {
    state: {
      admin_user?: NW_ADMIN_SYS_USER;
    };
  }
}
```

在写代码时，编辑器能提示到 **ctx.state.admin_user**  （√）, 引入的是import { Context } from '@midwayjs/koa'; 

现在升级成**V3.13.X**版本后，编辑器无法提示 （×）到state的定义了。
把import 改成 import { Context } from '@midwayjs/core';  才能有提示，但是又少了其他我要用到的属性。

是我的问题吗？
我记得在旧版本之前是能提示到state内的定义的，升级到V3.13.X后今天才发现没了。
![image](https://github.com/midwayjs/midway/assets/23348616/2eac5621-2488-49bb-bdda-9288696137bb)

### Midway Versions（Midway 版本）

>> Start to check your midway component version...    

✓ @midwayjs/faas-typings(not installed)
✓ @midwayjs/fc-starter(not installed)
✓ @midwayjs/serverless-http-parser(not installed)     
✓ @midwayjs/async-hooks-context-manager(3.13.5)       
✓ @midwayjs/axios(not installed)
✓ @midwayjs/bootstrap(3.13.5)
✓ @midwayjs/bull(3.13.5)   
✓ @midwayjs/bull-board(3.13.5)
✓ @midwayjs/cache(not installed)
✓ @midwayjs/captcha(not installed)
✓ @midwayjs/casbin(not installed)
✓ @midwayjs/casbin-redis-adapter(not installed)       
✓ @midwayjs/casbin-typeorm-adapter(not installed)     
✓ @midwayjs/code-dye(not installed)
✓ @midwayjs/consul(not installed)
✓ @midwayjs/core(3.13.5)   
✓ @midwayjs/cos(not installed)
✓ @midwayjs/cron(not installed)
✓ @midwayjs/cross-domain(3.13.5)
✓ @midwayjs/decorator(3.13.5)
✓ @midwayjs/etcd(not installed)
✓ @midwayjs/express-session(not installed)
✓ @midwayjs/faas(not installed)
✓ @midwayjs/grpc(not installed)
✓ @midwayjs/http-proxy(not installed)
✓ @midwayjs/i18n(3.13.5)
✓ @midwayjs/info(3.13.5)   
✓ @midwayjs/jwt(3.13.5)    
✓ @midwayjs/kafka(not installed)
✓ @midwayjs/leoric(not installed)
✓ @midwayjs/mikro(not installed)
✓ @midwayjs/mock(3.13.5)
✓ @midwayjs/mongoose(not installed)
✓ @midwayjs/oss(not installed)
✓ @midwayjs/otel(not installed)
✓ @midwayjs/passport(not installed)
✓ @midwayjs/process-agent(not installed)
✓ @midwayjs/prometheus(not installed)
✓ @midwayjs/prometheus-socket-io(not installed)       
✓ @midwayjs/rabbitmq(not installed)
✓ @midwayjs/redis(3.13.5)  
✓ @midwayjs/security(not installed)
✓ @midwayjs/sequelize(not installed)
✓ @midwayjs/session(3.13.5)✓ @midwayjs/socketio(not installed)
✓ @midwayjs/static-file(not installed)
✓ @midwayjs/swagger(not installed)
✓ @midwayjs/tablestore(not installed)
✓ @midwayjs/tags(not installed)
✓ @midwayjs/typegoose(not installed)
✓ @midwayjs/typeorm(3.13.5)✓ @midwayjs/upload(not installed)
✓ @midwayjs/validate(3.13.5)
✓ @midwayjs/version(3.13.5)✓ @midwayjs/view(not installed)
✓ @midwayjs/view-ejs(not installed)
✓ @midwayjs/view-nunjucks(not installed)
✓ @midwayjs/web(not installed)
✓ @midwayjs/express(not installed)
✓ @midwayjs/koa(3.13.5)    
✓ @midwayjs/ws(not installed)
**********************************************************************
>> Check complete, all versions are healthy.
**********************************************************************

**Comment by minefujiko9293** [comment]
@czy88840616

**Reply by czy88840616** [reply]
koa的 context 继承了 core 的，不过 state 是 koa/web/faas中扩展的，所以写的时候，定义的确要写到 koa 上。

**Reply by czy88840616** [reply, marked as answer]
现在 State 是个独立项，参考 https://github.com/midwayjs/midway/blob/main/packages/passport/index.d.ts#L13C10-L13C10

**Reply by minefujiko9293** [reply]
好的，哥。改好了，感谢
